cmd/fs-v1.go

```diff
@@ -232,76 +412,64 @@ func (fs fsObjects) GetObject(bucket, object string, offset int64, length int64,
 	}
 
 	// Stat the file to get file size.
-	fi, err := fs.storage.StatFile(bucket, object)
+	fi, err := os.Stat(preparePath(pathJoin(fs.fsPath, bucket, object)))
```

Comment:
There is no race. This entire thing is locked by namespace. This is why object level locking moved to handler layer.